main.c

```diff
@@ -2688,7 +2697,10 @@ create_directory (struct ovl_data *lo, int dirfd, const char *name, const struct
   if (ret < 0)
     goto out;
 
-  if (uid != lo->uid || gid != lo->gid || get_upper_layer (lo)->has_stat_override || get_upper_layer (lo)->has_privileged_stat_override)
+  if (!lo->squash_to_root
```

Comment:
could we move this check inside do_fchown?

Reply:
Done.

Would similar handling make sense for do_chown and do_fchownat?